src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -47,21 +53,18 @@ public ModelStateDictionary(int maxAllowedErrors)
         /// </summary>
         /// <param name="dictionary">The <see cref="ModelStateDictionary"/> to copy values from.</param>
         public ModelStateDictionary(ModelStateDictionary dictionary)
+            : this(dictionary?.MaxAllowedErrors ?? DefaultMaxAllowedErrors)
```

Comment:
My issue is the current code reads like it's intended to _support_ `dictionary == null`, not to give a correct error. So it's a bit icky even if "simpler". Shrug.